pkgs/by-name/eu/eudic/package.nix

```
@@ -0,0 +1,104 @@
+{ fetchurl
+, stdenv
+, autoPatchelfHook
+, makeWrapper
+, lib
+, makeDesktopItem
+, copyDesktopItems
+, fontconfig
+# Dependences of applications declaim
+, libnotify
+, libX11
+, libXScrnSaver
+, libXext
+, libXtst
+, libuuid
+, libsecret
+, xdg-utils
+, xdg-utils-cxx
+, at-spi2-atk
+# Dependences of autoPatchelfHook
+, gtk3
+, alsa-lib
+, libdrm
+, e2fsprogs
+, nss
+, libgpg-error
+, libjack2
+, mesa
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "eudic";
+  version = "13.5.2";
+
+  src = fetchurl {
+    url = "https://www.eudic.net/download/eudic.deb?v=${finalAttrs.version}";
+    hash = "sha256-UPkDRaqWF/oydH6AMo3t3PUT5VU961EPLcFb5XwOXVs=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  # Depends: libgtk-3-0, libnotify4, libnss3, libxss1, libxtst6, xdg-utils, libatspi2.0-0, libuuid1, libsecret-1-0
+  buildInputs = [
+    libnotify
+    libX11
+    libXScrnSaver
+    libXext
+    libXtst
+    libuuid
+    libsecret
+    xdg-utils
+    xdg-utils-cxx
+    at-spi2-atk
+    # Dependences base on autoPatchelfHook and nix-locate
+    gtk3                # libEGL.so.1 libfreetype.so.6 libfontconfig.so.1
+    alsa-lib            # libasound.so.2
+    e2fsprogs           # libcom_err
+    nss                 # libnss3.so libnssutils3.so
+    libgpg-error        # libgpg-error.so
+    libjack2            # libjack2.so.0
+    mesa                # libgbm.so.1
+  ];
+
+  unpackPhase = ''
+    ar x $src
+    tar xf data.tar.xz
+  '';
+
+  # After unpackPhase
+  # File is /usr/share/{Application, eusoft-eudic, icons, pixmaps}
+  installPhase = ''
+    runHook preInstall
+
+    # Copy & Paste Original File
+    echo $out
+    mkdir -p $out
+    cp -r usr/* $out/
+
+    # Build Wrapper
+    makeWrapper $out/share/eusoft-eudic/eudic $out/bin/eudic \
+      --argv0 "eudic" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath finalAttrs.buildInputs}"
+
+    install -D -m 444 $out/share/applications/eusoft-eudic.desktop -t $out/bin
+
+    runHook postInstall
```

```suggestion
    runHook postInstall
```
desktop files to not belong there